File: MyJournal/app/src/main/java/com/smart/weather/module/journal/viewmodel/JournalViewModel.java
package com.smart.weather.module.journal.viewmodel;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.smart.weather.module.journal.repository.JournalRepository;
import com.smart.weather.module.journal.repository.JournalRepositoryImpl;
import com.smart.weather.module.write.bean.JournalBeanDBBean;

import java.util.List;

import io.realm.RealmResults;

/**
 * @author guandongchen
 * @date 26/03/2018
 */

public class JournalViewModel extends ViewModel implements JournalRepository{

    private JournalRepository journalRepository;

    private MutableLiveData<RealmResults<JournalBeanDBBean>> listMutableLiveData;


    public JournalViewModel() {

        journalRepository = new JournalRepositoryImpl();

        if (this.listMutableLiveData == null) {
            // ViewModel is created per Fragment so
            // we know the userId won't change
            listMutableLiveData = new MutableLiveData<>();
        }
        listMutableLiveData = journalRepository.getLiveDataJournalBeans();

    }


    @Override
    public MutableLiveData<RealmResults<JournalBeanDBBean>> getLiveDataJournalBeans() {
        return journalRepository.getLiveDataJournalBeans();
    }

    @Override
    public List<JournalBeanDBBean> getJournalBeans() {
        return journalRepository.getJournalBeans();
    }

    @Override
    public void deleteJournal(JournalBeanDBBean journalBeanDBBean) {
        journalRepository.deleteJournal(journalBeanDBBean);
    }
}

File: MyJournal/app/src/main/java/com/smart/journal/module/journal/viewmodel/JournalViewModel.kt
package com.smart.journal.module.journal.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.blankj.utilcode.util.FileUtils
import com.smart.journal.contants.Contancts
import com.smart.journal.db.entity.JournalBeanDBBean
import com.smart.journal.module.journal.repository.JournalRepository
import com.smart.journal.module.journal.repository.JournalRepositoryImpl
import java.util.logging.Logger


/**
 * @author guandongchen
 * @date 26/03/2018
 */

class JournalViewModel : ViewModel() {

    private val journalRepository: JournalRepository

    init {

        journalRepository = JournalRepositoryImpl()

    }

    fun deleteJournal(journalBeanDBBean: JournalBeanDBBean) {
        journalRepository.deleteJournal(journalBeanDBBean)
        if (journalBeanDBBean.content != null) {

            val contents = journalBeanDBBean.content!!.split(Contancts.FILE_TYPE_SPLIT.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

            for (content in contents) {
                //查看内容里是不是包含图片
                if (content.startsWith(Contancts.FILE_TYPE_IMAGE)) {
                    //删除本地图片
                    var imagePath = content.replace(Contancts.FILE_TYPE_IMAGE,"")
                    com.orhanobut.logger.Logger.d(imagePath)
                    FileUtils.delete(imagePath!!)
                }

            }
        }
    }


    fun searchJournalByKeyWord(keyWord: String): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.searchJournalByKeyWord(keyWord)
    }

    fun searchJournalByTag(tagName: String): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.searchJournalByTag(tagName)
    }

    fun searchJournalByFavourite(tagName: String): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.searchJournalByTag(tagName)
    }

    fun searchJournalByLocationName(locationName: String): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.searchJournalByLocationName(locationName)
    }


    fun getJournalBeans(): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.getJournalBeans()
    }

    fun searchByNoteBookName(noteBookName: String): LiveData<List<JournalBeanDBBean>> {
        return journalRepository.searchByNoteBookName(noteBookName)
    }

    public override fun onCleared() {
        journalRepository.onLiveDataCleared()
        super.onCleared()
    }
}
